### PR TITLE
Set environment variables in executor

### PR DESCRIPTION
**Pull request by mstajdohar**

### CODE REVIEW

**Comment by mstajdohar** [comment]
New configuration environment variables: `RESOLWE_API_USERNAME` and `RESOLWE_API_PASSWORD`

File: docs/CHANGELOG.rst [hunk 1]
```diff
@@ -28,6 +28,7 @@ Added
 - Archive file process
 - File upload processes
 - Resolwe process tests
+- Add RESOLWE_API_USERNAME and RESOLWE_API_PASSWORD configuration options
```
Comment (dblenkus) [inline docs/CHANGELOG.rst]: Describe where and why settings were added.

**Comment by kostko** [comment]
> How do you think this would fit into the resources constrains you are working on?

Hm, so these are both security tokens, which are VERY sensitive. AFAIK, each process that needs them should be explicitly granted permissions to have these tokens (e.g., via a special `api` resource, which should also specify which permissions should be granted).

IMO the best approach would be to create a temporary security token before running the process, which is then revoked immediately after the process exits.

**Comment by mstajdohar** [comment]
Let's not over-engineer here. The goal should be that we do not have to pass `resdk` credentials to the nodes at all. When we will introduce an upgrade of the workflow language (to support the demultiplexing use case) we should remove this _temporary hack_.

**Comment by kostko** [comment]
If this is a temporary hack, please add `XXX: Temporary hack, remove when we support demultiplexing.` in appropriate places.

**Comment by mstajdohar** [comment]
> please add XXX: Temporary hack, remove when we support demultiplexing

Will do. 

> But note that this hack introduces a potential security risk.

This hack does not introduce a security risk if not used. We should not and will not use this on public servers.

**Comment by mstajdohar** [comment]
With this change even the `RESOLWE_HOST_URL` is not needed anymore. We should remove it when we update the install scripts.

File: docs/CHANGELOG.rst [hunk 2]
```diff
@@ -28,6 +28,7 @@ Added
 - Archive file process
 - File upload processes
 - Resolwe process tests
+- Add SET_ENV setting to set environment variables in executor
```
Comment (tjanez) [inline docs/CHANGELOG.rst]: I would change this to something like:
```
``FLOW_EXECUTOR['SET_ENV']`` setting that allows setting environment variables in executor
```

File: resolwe/flow/tests/test_env_vars.py [hunk 3]
```diff
@@ -12,33 +15,42 @@ class EnvVarsTest(TestCase):
...
-            contributor=self.contributor,
-            type='test:data:envvars:',
-            input_schema=[],
-            output_schema=[
-                {'name': 'resolweapihost', 'type': 'basic:string:'},
-            ],
-            run={
-                'language': 'bash',
-                'program': """
+        flow_executor = copy.copy(getattr(settings, 'FLOW_EXECUTOR', {}))
```
Comment (tjanez) [inline resolwe/flow/tests/test_env_vars.py]: After #220 is merged, this will be much nicer to write with the `@with_custom_executor` decorator.

**Pull request merged**